subcommand/consul-sidecar/command_test.go

```diff
@@ -79,6 +79,67 @@ func TestRunSignalHandlingRegistration(t *testing.T) {
 	}
 }
 
+func TestRunSignalHandlingMetricsOnly(t *testing.T) {
+	cases := map[string]os.Signal{
+		"SIGINT":  syscall.SIGINT,
+		"SIGTERM": syscall.SIGTERM,
+	}
+	for name, signal := range cases {
+		t.Run(name, func(t *testing.T) {
+			ui := cli.NewMockUi()
+			cmd := Command{
+				UI: ui,
+			}
+
+			randomPorts := freeport.MustTake(1)
+			// Run async because we need to kill it when the test is over.
+			exitChan := runCommandAsynchronously(&cmd, []string{
+				"-enable-service-registration=false",
+				"-enable-metrics-merging=true",
+				"-merged-metrics-port", fmt.Sprint(randomPorts[0]),
+				"-service-metrics-port", "8080",
+				"-service-metrics-path", "/metrics",
+			})
+
+			// Keep an open connection to the server by continuously sending bytes
+			// on the connection so it will have to be drained.
+			var conn net.Conn
+			var err error
+			retry.Run(t, func(r *retry.R) {
+				conn, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", randomPorts[0]))
+				if err != nil {
+					require.NoError(r, err)
+				}
+			})
+			go func() {
+				for {
+					_, err := conn.Write([]byte("hello"))
+					if err != nil {
+						break
+					}
+				}
+			}()
```

Comment:
yes it will, since it will not be able to write to the connection, so it will break out of the for loop, exiting the goroutine so it can be cleaned up. It's a great idea to add a comment there, thanks!